addons/binding/org.openhab.binding.sensebox/src/main/java/org/openhab/binding/sensebox/handler/SenseBoxHandler.java

```diff
@@ -0,0 +1,373 @@
+
+import static org.openhab.binding.sensebox.SenseBoxBindingConstants.*;
+
+import java.util.List;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+import java.io.IOException;
+import java.net.URL;
+import java.net.URLConnection;
+
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.lang.StringUtils;
+import org.eclipse.smarthome.core.library.types.DateTimeType;
+import org.eclipse.smarthome.core.library.types.DecimalType;
+import org.eclipse.smarthome.core.library.types.PointType;
+import org.eclipse.smarthome.core.library.types.StringType;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.eclipse.smarthome.core.types.State;
+import org.eclipse.smarthome.core.types.UnDefType;
+import org.openhab.binding.sensebox.config.SenseBoxConfiguration;
+import org.openhab.binding.sensebox.internal.ExpiringCache;
+import org.openhab.binding.sensebox.model.SenseBoxData;
+import org.openhab.binding.sensebox.model.SenseBoxDescriptor;
+import org.openhab.binding.sensebox.model.SenseBoxLoc;
+import org.openhab.binding.sensebox.model.SenseBoxLocation;
+import org.openhab.binding.sensebox.model.SenseBoxSensor;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+
+/**
+ * The {@link SenseBoxHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Hakan Tandogan - Initial contribution
+ */
+
+public class SenseBoxHandler extends BaseThingHandler {
+    private Logger logger = LoggerFactory.getLogger(SenseBoxHandler.class);
+
+    protected SenseBoxConfiguration thingConfiguration;
+
+    private Gson gson = new Gson();
+
+    private SenseBoxData data = new SenseBoxData();
+
+    ScheduledFuture<?> refreshJob;
+
+    private final int CACHE_EXPIRY = 10 * 1000; // 10s
+
+    public SenseBoxHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("Start initializing!");
+
+        thingConfiguration = getConfigAs(SenseBoxConfiguration.class);
+        String thingUid = getThing().getUID().toString();
+        thingConfiguration.setThingUid(thingUid);
+
+        logger.debug("Thing Configuration {} initialized {}", thingConfiguration.getThingUid(),
+                thingConfiguration.getSenseBoxId());
+
+        String offlineReason = "";
+        boolean validConfig = true;
+
+        if (StringUtils.trimToNull(thingConfiguration.getSenseBoxId()) == null) {
+            offlineReason = "senseBox ID is mandatory and must be configured";
+            logger.error("{}, disabling thing '{}'", offlineReason, thingUid);
```

Comment:
Check our logging guidelines (E.5): http://docs.openhab.org/developers/development/guidelines.html#e-logging
You should not do logging, but set the ThingStatus instead.